tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -4003,9 +4043,11 @@ Result D3D12Device::createBuffer(const D3D12_RESOURCE_DESC& resourceDesc, const
         heapProps.CreationNodeMask = 1;
         heapProps.VisibleNodeMask = 1;
 
+        D3D12_HEAP_FLAGS flags = isShared ? D3D12_HEAP_FLAG_SHARED : D3D12_HEAP_FLAG_NONE;
```

Comment:
Coding style thing (this is *not* a "coding convention" rule or anything, just a thing I've learned over years): very often when you have flags like this, the reality is that you end up wanting to add/include different flags under different conditions and trying to write it all as a big conditional with `?:` actually harms readability.

My strong preference when I end up having to write something like this is to break it down into multiple lines such that it can be a lot more future-proof:

```
D3D12_HEAP_FLAGS flags = D3D12_HEAP_FLAG_NONE;
if(isShared) flags |= D3D12_HEAP_FLAG_SHARED;
```

For now the effect is the exact same, but the difference is that the alternative way of writing it can easily handle a future where we find out that sometimes we need to allocate heaps with the new "nifty" flag:

```
if(isNifty) flags |= D3D12_HEAP_FLAG_NIFTY;
```

Adding support for a new flag becomes a one-line change that doesn't disrupt the existing logic, instead of requiring the next programmer who touches this to sweat about breaking apart a `?:` thing and wonder what to replace it with. I always like writing code that will minimize the diffs I have to write in the future.

Reply:
I'm pretty sure I did this the first time I added this flag because that's how it was done everywhere else I looked and then forgot on the redo. Makes sense though.